numpy/_core/src/multiarray/dtypemeta.h

```diff
@@ -291,6 +299,37 @@ PyArray_SETITEM(PyArrayObject *arr, char *itemptr, PyObject *v)
                 Py_XSETREF(descr, _new_);  \
         } while(0)
 
+static inline int
+PyArray_GetSortFunction(PyArray_Descr *descr, 
+    NPY_SORTKIND which, int descending, PyArray_SortFunc **out_sort)
+{
+    if (NPY_DT_SLOTS(NPY_DTYPE(descr))->get_sort_function == NULL) {
+        return -1;
```

Comment:
This needs to set an error (TypeError or DTypeTypeError, which is defined somewhere I think.)

(An error here will make sense after you move the fallback logic into a default slot filling.  Or you could have the default slot raise an error, that is also completely fine.)